docs/developer/advanced/sharing-saved-objects.asciidoc

```diff
@@ -0,0 +1,437 @@
+<<xpack-spaces, space>>) or _global_ (existing in all spaces), there was no in-between. As of the 7.12 release, {kib} now supports two
+additional types of objects:
+
+|======================================================================================================
+|                     | *Where it exists* | *Object IDs*         | *Registered as:*
+| Global              | All spaces        | Globally unique      | `namespaceType: 'agnostic'`
+| Isolated            | 1 space           | Unique in each space | `namespaceType: 'single'`
+| (NEW) Share-capable | 1 space           | Globally unique      | `namespaceType: 'multiple-isolated'`
+| (NEW) Shareable     | 1 or more spaces  | Globally unique      | `namespaceType: 'multiple'`
+|======================================================================================================
+
+Ideally, most types of objects in {kib} will eventually be _shareable_; however, we have also introduced
+<<sharing-saved-objects-faq-share-capable-vs-shareable,_share-capable_ objects>> as a stepping stone for plugin developers to fully support
+this feature.
+
+[[sharing-saved-objects-breaking-changes]]
+=== Breaking changes
+
+To implement this feature, we had to make a key change to how objects are serialized into raw {es} documents. As a result,
+<<sharing-saved-objects-faq-changing-object-ids,some existing object IDs need to be changed>>, and this will cause some breaking changes to
+the way that consumers (plugin developers) interact with objects. We have implemented mitigations so that *these changes will not affect
+end-users _if_ consumers implement the required steps below.*
+
+Existing, isolated object types will need to go through a special _conversion process_ to become share-capable upon upgrading {kib} to
+version 8.0. Once objects are converted, they can easily be switched to become fully shareable in any future release. This conversion will
+change the IDs of any existing objects that are not in the Default space. Changing object IDs itself has several knock-on effects:
+
+* Nonstandard links to other objects can break - _mitigated by step 1_
+* "Deep link" pages (URLs) to objects can break - _mitigated by steps 2/3_
+* Encrypted objects may not be able to be decrypted - _mitigated by step 5_
+
+*To be perfectly clear: these effects will all be mitigated _if and only if_ you follow the steps below!*
+
+[[sharing-saved-objects-dev-flowchart]]
+=== Developer Flowchart
+
+If you're still reading this page, you're probably developing a {kib} plugin that registers an object type, and you want to know what steps
+you need to take to prepare for the 8.0 release and mitigate any breaking changes! Depending on how you are using saved objects, you may
+need to take up to 5 steps, which are detailed in separate sections below. Refer to this flowchart:
+
+image::images/sharing-saved-objects-dev-flowchart.png["Sharing Saved Objects developer flowchart"]
```

Comment:
Yeah, the idea here was to have these "thumbnails" provide a bit of a visual aid so when you scroll down to the steps below, you see the larger versions of the thumbnails, and it's easier to associate where you are at in the flowchart.

Reply:
I see, maybe it's just me then, when I scrolled down to the actual steps I already forgot how these thumbnails looked like since I couldn't recognize them in the first place :see_no_evil: Feel free to keep it as is then!